Added support for timedelta expiration like redis-py

Pydes-boop opened the pull request:
redis-py already supports EX in int as seconds since epoch or as timedelta, this PR exposes the timedelta possibility for expiration from redis-py to redis-dict.

Additionally i added some tests to make sure it is all covered but those could potentially be left out because their functionality might be tested by redis-py already.

Attumm commented on tests.py:
```diff
@@ -808,19 +809,31 @@ def test_chain_del_2(self):
+        actual_ttl = self.redisdb.ttl('{}:foobar1'.format(TEST_NAMESPACE_PREFIX))
         self.assertAlmostEqual(3600, actual_ttl, delta=2)
+        actual_ttl = self.redisdb.ttl('{}:foobar2'.format(TEST_NAMESPACE_PREFIX))
+        self.assertAlmostEqual(2700, actual_ttl, delta=2)
```
Great job! However, let's distinguish between the current test and the new functionality tests. The reason for doing so is as follows:
Each test plays a crucial role in ensuring the stability of the project. By creating separate tests for every new functionality, we will have the flexibility to add and remove features in the future without having to modify our unit tests.
This approach also makes each Pull Request more self-contained.

Could you consider the code below?
```
  def test_expire_context_timedelta(self):
         """
         Test adding keys with an expire value by using the contextmanager.
         With timedelta as argument.
         """
         minute_in_seconds = 60
         hour_in_seconds = 60 * 60
         
         one_hour_timedelta = timedelta(hours=1)
         with self.r.expire_at(one_hour_timedelta):
             self.r['one_hour'] = 'one_hour'
             
         one_minute = timedelta(minutes=1)
         with self.r.expire_at(one_hour_timedelta):
             self.r['one_minute'] = 'one_minute'

         actual_ttl = self.redisdb.ttl('{}:one_hour'.format(TEST_NAMESPACE_PREFIX))
         self.assertAlmostEqual(hour_in_seconds, actual_ttl, delta=2)
         
         actual_ttl = self.redisdb.ttl('{}:one_minute'.format(TEST_NAMESPACE_PREFIX))
         self.assertAlmostEqual(minute_in_seconds,  actual_ttl, delta=2)
  ```

Attumm commented on tests.py:
```diff
@@ -808,19 +809,31 @@ def test_chain_del_2(self):
+        actual_ttl = self.redisdb.ttl('{}:foobar2'.format(TEST_NAMESPACE_PREFIX))
+        self.assertAlmostEqual(2700, actual_ttl, delta=2)
 
     def test_expire_keyword(self):
```
Same as the other comment.

Attumm commented on redis_dict.py:
```diff
@@ -640,12 +641,12 @@ def chain_del(self, iterable: List[str]) -> None:
 
         Args:
-            sec_epoch (int): The expiration time in Unix timestamp format.
+            sec_epoch (int, timedelta): The expiration time in Unix timestamp format.
```
Let's update the docstring:

```
sec_epoch (int, timedelta): The expiration duration is set using either an integer or a timedelta 
```

Attumm reviewed (changes requested):
Great job with the PR, however, there are a few points.

Firstly, could we add new tests, instead of modifying the existing ones?

By adding new tests, we can categorize each feature with its own tests. This approach will contribute to the project's stability. Moreover, because each feature will have its own tests, it will reduce maintenance and cognitive load during the debugging of failed unit tests.

Additionally, updating the docstring to reflect that a new type could be passed would be beneficial.

Pydes-boop commented:
Thanks for the feedback! I changed the structure of your suggested codesnippet a bit as i found it more readable but otherwise all should be fixed and added as requested in the latest commit.

Let me know if there is anything else.
